### PR TITLE
Let the user specify which monitor OPENRNDR uses

### LINKED ISSUE #113: Allow specifying monitor for full screen window
**Is your feature request related to a problem?**
Users with two monitors can't specify in which one to show the full screen window

**Describe the solution you'd like**
An integer in `configuration` to specify the number of the monitor to use.

### LINKED ISSUE #304: Bug report: Multiple Displays Fullscreen Automatically Minimises on Focus Loss
### Operating System

Mac

### OPENRNDR version

0.4.0

### Java version (if applicable)

_No response_

### Describe the bug

If a fullscreen app with open render is created, then on a system with multiple displays, the window is automatically minimised if focus is lost.

I have tried various things like setting: `glfwWindowHint(GLFW_AUTO_ICONIFY, GLFW_FALSE)`

The other problem is that in fullscreen mode, the app does not exist as a native macos window. If you swipe up on the trackpad to open `App Exposé` then the openrndr window disappears and cannot be seen.

### Steps to reproduce the bug

Create the fullscreen app:
```kotlin
fun main() = application {
    configure {
        fullscreen = Fullscreen.CURRENT_DISPLAY_MODE
    }
}
```

Then try change focus to the second display that is not currently active.

### PR DESCRIPTION
**Pull request by Vechro**
This is a bit trickier than I first assumed. It's a complicated feature to test so I will initially be publishing this PR as a draft until I'm more confident it can handle all the edge cases.
Fixes #113. Fixes #304.

### CODE REVIEW

**Comment by Vechro** [comment]
Would love to hear how it fares on macOS and Linux distributions. I've only been able to test this on Windows.

**Comment by hamoid** [comment]
How and what should we test? I did

```
git fetch upstream pull/303/head:monitors
git checkout monitors
./gradlew publishToMavenLocal -Prelease.version=0.5.1-SNAPSHOT
```

Then tried this program
```
import org.openrndr.application

fun main() = application {
    configure {
        monitor = 0
    }
    program {
        extend {
            drawer.circle(drawer.bounds.center, 200.0)
        }
    }
}
```

`monitor` 0 and 1 display the program on the same window as the IDE, with 2 it fails to run.

What can I try next? Maybe you can add debug messages to the code and I see what comes out on Linux?

**Comment by hamoid** [comment]
I wonder if it's because I'm using a tiling window manager (i3wm). I'll try with the default.

**Comment by hamoid** [comment]
That was it. With the default Ubuntu wm it works as expected and I can choose the monitor with 0 or 1, both normal and full screen.

**Comment by Vechro** [comment]
That's interesting, and yes I should have specified what to test. I've mainly been looking
1. Whether `monitor` sets the window on the specified monitor.
2. Does fullscreen work as expected?
3. Does specifying the window position work as expected?
4. If the window position is not specified, does it properly center the window on the specified monitor?

**Comment by hamoid** [comment]
I didn't try 3, the others did work in the Ubuntu wm. I'll try 3 this evening (too lazy to close all apps, log out and log in again, then reverse).

Meanwhile I was trying this

```
        run {
            println("monitors!!!")
            val a = glfwGetMonitors()?.get(0)!!
            val b = glfwGetMonitors()?.get(1)!!
            val x = FloatArray(1)
            val y = FloatArray(1)
            glfwGetMonitorContentScale(a, x, y)
            println("MON 0: ${x[0]} ${y[0]}")
            glfwGetMonitorContentScale(b, x, y)
            println("MON 1: ${x[0]} ${y[0]}")

            val mode0 = glfwGetVideoMode(a)!!
            println("MON 0: ${mode0.width()} x ${mode0.height()}")
            val mode1 = glfwGetVideoMode(a)!!
            println("MON 1: ${mode1.width()} x ${mode1.height()}")
        }
```

which prints the expected values even if it doesn't work as expected in i3wm. At least it does get the right screen resolutions.
```
monitors!!!
MON 0: 1.0 1.0
MON 1: 1.0 1.0
MON 0: 1920 x 1200
MON 1: 1680 x 1050
```

**Comment by hamoid** [comment]
From GLFW.java:2026
image

Can this affect in my case? is it an "initially hidden window"?

**Comment by hamoid** [comment]
Trying to figure out what OPENRNDR knows about my window position...
```
    program {
        extend {
            drawer.circle(drawer.bounds.center, 200.0)
        }
        window.moved.listen {
            println(it.position)
        }
        window.focused.listen {
            println("focused")
        }
        window.unfocused.listen {
            println("unfocused")
        }
    }
```
I see that `focused` and `unfocused` events are triggered, but not `moved`. Anyway, I'm used to things working different in my system. At least it works fine with the common wm.

**Comment by Vechro** [comment]
> From GLFW.java:2026 image
> 
> Can this affect in my case? is it an "initially hidden window"?

That could very well be the case, you could try changing [this window hint](https://github.com/Vechro/openrndr/blob/8a52ffced6457b4280a3cfc7c50735690facae59/openrndr-jvm/openrndr-gl3/src/jvmMain/kotlin/org/openrndr/internal/gl3/ApplicationGLFWGL3.kt#L179) and see if that fixes it. I wouldn't change it in the PR because I feel like it would introduce some visual glitches for everyone but it could verify the cause of the problem.

**Comment by hamoid** [comment]
Leaving this note here so it can be found together with the above:

Changing `glfwWindowHint(GLFW_VISIBLE, GLFW_FALSE)` to `GLFW_TRUE` makes the monitor selection work in i3. The issue then might be how to customize a setting only in this case? The [env var](https://twitter.com/hamoid/status/1528777986090143745) `DESKTOP_SESSION` contains `i3` in my case.

But instead of being smart and trying to detect window managers, maybe it's safer to allow to specify in some way `tiling window manager` present? Because there's not only i3, there are others. And monitor selection is not the only thing affected by using a tiling wm. Something to think about some day.

File: openrndr-application/src/jvmMain/kotlin/org/openrndr/Application.kt [hunk 1]
```diff
@@ -87,20 +79,39 @@ actual abstract class Application {
...
-                Application::class.java.classLoader.loadClass("org.openrndr.internal.gl3.ApplicationGLFWGL3")
-            else
-                Application::class.java.classLoader.loadClass("org.openrndr.internal.gl3.ApplicationEGLGL3")
+            val applicationProperty: String? = System.getProperty("org.openrndr.application")
+
+            @Suppress("KotlinConstantConditions")
+            return when (applicationProperty) {
+                null, "", "ApplicationGLFW" -> Application::class.java.classLoader.loadClass("org.openrndr.internal.gl3.ApplicationGLFWGL3")
+                "ApplicationEGL" -> Application::class.java.classLoader.loadClass("org.openrndr.internal.gl3.ApplicationEGLGL3")
+                else -> throw IllegalArgumentException("Unknown value '${applicationProperty}' provided for org.openrndr.application")
```
Comment (Vechro) [inline openrndr-application/src/jvmMain/kotlin/org/openrndr/Application.kt:88]: Are these names intuitive for the user? Should we instead simply use "GLFW" and "EGL"?
